src/lib/utils/calc-salt.ts

```diff
@@ -0,0 +1,13 @@
+import { ethers } from 'ethers';
+
+export default function calculateSaltFromAddress(seedConstant: string, address: string) {
+  const hash = ethers.keccak256(
+    ethers.AbiCoder.defaultAbiCoder().encode(['string'], [seedConstant + address]),
+  );
+  const addressBigInt = ethers.toBigInt('0x' + hash.slice(26));
+
+  const randomBytes = ethers.randomBytes(32);
```

Comment:
This definitely introduces randomness now 👍
But, why not simplifying to something simple like:

```ts
export default function generateRandomNumber() {
  const randomBytes = ethers.randomBytes(32);
  return BigInt(ethers.hexlify(randomBytes)) & BigInt('0xFFFFFFFFFFFFFFFF');
}
```

Reply:
No reason at all. Just not trying to shake the tree too much because I don't know what constraints the seed should have. Will update!